### ISSUE TITLE
Cannot detect enum method in protocol

### ISSUE THREAD

**Issue by jie-huang** [PyCQA/pylint#7729]
### Bug description

test.py
```
from enum import Enum
from typing import Protocol

class WeekDay(Enum):
    Monday = 'monday'

class Event(Protocol):
    def weekday(self) -> WeekDay: ...
    
    def hours(self) -> int: ...


class EventMix(Event):
    def do(self):
        print(f'weekday={self.weekday.value}')
        print(f'hours={self.hours}')

class MyEvent(EventMix):
    def __init__(self) -> None:
        self.weekday = WeekDay.Monday

my_event = MyEvent()
my_event.do()
```

pylint always reports
`Method 'weekday' has no 'value' memberpylint(no-member)`

### Command used

```shell
python test.py
pylint test.py
```

### Pylint output

```shell
E1101: Method 'weekday' has no 'value' member (no-member)
```

### Expected behavior

It should not report error because it is correct to run

### Pylint version

```shell
pylint 2.15.5
astroid 2.12.12
Python 3.10.5 (v3.10.5:f377153967, Jun  6 2022, 12:36:10) [Clang 13.0.0 (clang-1300.0.29.30)]
```

**Comment by mbyrnepr2** [comment]
Thanks for the report. I think it isn't an issue with Enum or Protocol per-se.
Pylint doesn't know that parent class is referencing a name defined in the child class; here's a simplification:
```python
class Tomato:
    def print_me(self):
        print(self.sweetness)  # 3:14: E1101: Instance of 'Tomato' has no 'sweetness' member (no-member)

class SweetTomato(Tomato):
    def __init__(self):
        self.sweetness = 42


SweetTomato().print_me()
```

**Comment by DanielNoord** [comment]
Isn't the issue with the `Protocol` as in: it says that `sweetness` or `weekday` should be defined?
Then again, at runtime the `Protocol` has no effect so in theory this would still crash in the parent class and therefore the message is valid?

**Comment by jie-huang** [comment]
There should be a bug in pylint.

Since weekday is a **method** in Event, pylint reports method has no 'value' member. That makes sense.
But, it **should** report error when `self.weekday = WeekDay.Monday`.

**Comment by mbyrnepr2** [comment]
I think we can agree that it is valid Python but we need to decide if we want to suppress this message for a scenario like this or suggest something else. The `mypy` output is interesting:
```
example.py:16: error: Cannot assign to a method  [assignment]
example.py:16: error: Incompatible types in assignment (expression has type "WeekDay", variable has type "Callable[[], WeekDay]")  [assignment]
example.py:18: error: Cannot instantiate abstract class "MyEvent" with abstract attribute "weekday"  [abstract]
```
